core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -249,6 +266,15 @@ private ClientTransport obtainActiveTransport() {
         // start() must return before we set activeTransport, since activeTransport is accessed
         // without a lock.
         activeTransport = newActiveTransport;
+        backoff = null;
+      } else {
```

Comment:
@ejona86 are you suggesting that this code be located elsewhere? If so, where ... in `TransportListener`?